### PR TITLE
Add rerun script

### PR DESCRIPTION
**Pull request by michaelosthege**
This PR adds `scripts/rerun.py` and applies it to the BEST notebook.
It can be used from the command line or from another Python script to apply API migrations, re-run notebooks and optionally commit to a specific branch and push it to a remote.

The replacements are the ones @twiecki wrote for #262.

For example, this is how we can apply it for the example scripts:
```
import rerun

for fp in ["example1.py", "example2.py"]:
    assert rerun.apply_replacements("some_example.py")
    assert rerun.commit(fp, "update-some-example")
assert rerun.push("update-some-example", "myremotename")
```

@twiecki @OriolAbril I think we should chunk the work and, for example, have one PR that updates only the example scripts like shown above.
If needed those PRs can then adapt the `REPLACEMENTS`.

### CODE REVIEW

File: scripts/rerun.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import logging
+import pathlib
+import subprocess
+
+
+logging.basicConfig(level=logging.INFO)
+_log = logging.getLogger(__file__)
+DP_REPO = pathlib.Path(__file__).absolute().parent.parent
+
+REPLACEMENTS = {
```
Comment (twiecki) [inline scripts/rerun.py:27]: The order matters here.

Comment (michaelosthege) [reply scripts/rerun.py:27]: Ah, so that's why there's the `stundenat`→`studentt`!
I though you were correcting typos there..

**Comment by OriolAbril** [comment]
This is related to #52, so we keep track of everything. I will also create a `v4 (auto)` column in the notebook tracker to take into account that they have been updated but might still need code updated to show v4 best practices, loops of scalar distributions converted to correct use of the size argument, removing now unnnecessary model factories things like that. 

The PR LGTM, and merging this and aiming for multiple PRs also allows anyone to help.

**Review by OriolAbril — APPROVED** [review]

**Comment by OriolAbril** [comment]
#265 updates the BEST example on both code and style, can we autoupdate another notebook here to avoid merge issues?

**Comment by twiecki** [comment]
What happened to the AB testing NB?

**Comment by michaelosthege** [comment]
> What happened to the AB testing NB?

I undid that commit to avoid conflicts, as requested by
https://github.com/pymc-devs/pymc-examples/pull/264#issuecomment-1008267679

The notebook I re-ran now needed a manual intervention, but I got annoyed by automated re-runs not working, so I just fixed it.
Most breaking changes that need manual intervention happen around working with `InferenceData` instead of `MultiTrace`.

**Comment by OriolAbril** [comment]
related to #110

File: scripts/rerun.py (hunk 1 above)
Comment (twiecki) [reply scripts/rerun.py:27]: Should we not make this an OrderedDict or can we rely on `dict` being ordered now?

File: scripts/rerun.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    "az.from_pymc3": "pm.to_inference_data",
+    "arviz.from_pymc3": "pm.to_inference_data",
+    "pymc3": "pymc",
+    "PyMC3": "PyMC",
+    "pymc3": "pymc",
```
Comment (twiecki) [inline scripts/rerun.py:32]: these occur twice.

**Pull request merged**